tutorials/cp-cf-understand-application-lifecycle/cp-cf-understand-application-lifecycle.md

```diff
@@ -0,0 +1,97 @@
+
+The uploading phase begins as follows:
+
+!![Uploading](uploading-phase.png)
+
+After that, the staging phase begins. You can see the buildpacks being downloaded (since a specific buildpack is not specified it downloads all of them):
+
+!![Staging Begins](staging-phase.png)
+
+You can also see when the droplet get uploaded:
+
+!![Droplet uploaded](droplet-uploaded.png)
+
+And finally you see the starting phase:
+
+!![Application starting](starting-phase.png)
+
+> If you run `cf push` for an existing application, _all instances of that application are stopped_ and new instances are started with the code you just pushed. Users that try to access the app during this time may get various HTTP error messages, such as "404" ("file not found") and others, while the application is redeployed. The Cloud Foundry API `v3` support the so called [Rolling App Deployments](https://docs.cloudfoundry.org/devguide/deploy-apps/rolling-deploy.html) with `cf push <app-name> --strategy rolling` which allows to push an app without downtime. The API `v3` are also supported on SAP BTP.
+
+[DONE]
+[ACCORDION-END]
+
+[ACCORDION-BEGIN [Step 2: ](The Three R's - Restarting, Re-staging, and Re-pushing)]
+
+Knowing when to re-stage, restart, or re-push is useful in properly maintaining your application with minimal downtime or even without a downtime. All this can be a little confusing, so here's a quick summary and overview:
+
+- _Re-push_ - if there is new or changed code. When application code changes, the developer has to `push` that new/changed code to Cloud Foundry before CF can do anything with it. A quick `cf push <app-name>` copies the most recent code up to your Cloud Foundry environment so that the changes can be compiled and deployed. More detailed information on this process can be seen in the section above "CF Push".
+
+- _Re-stage_ - when there is updated configuration or a change in app environment which has an effect for the staging process, but your app code's the same. If the application code has not changed, but the environment or configuration has, then doing a re-stage (`cf restage <app-name>`) will get CF to populate those changes down to running instances of your app. Check out the CF docs on [re-staging your app](https://docs.cloudfoundry.org/devguide/deploy-apps/start-restart-restage.html#restage) for more information.
+
+- _Re-start_ - app instances when there are changed environment variables or service bindings. Additionally, `cf restart <app-name>` is useful when you want to quickly recycle application processes and get an application back to a fresh/known state. If you're interested in more details check out the [official documentation](https://docs.cloudfoundry.org/devguide/deploy-apps/start-restart-restage.html#restart).
+
+
+[VALIDATE_1]
+[ACCORDION-END]
+
+[ACCORDION-BEGIN [Step 4: ](Crashes, Shutdowns, and Evacuation)]
```

Comment:
Action verb, "Manage crashes, shutdowns, and evacuations"  These are suggestions by the way.

Reply:
This feedback has been applied.